azure-kusto-data/azure/kusto/data/_cloud_settings.py

```diff
@@ -87,9 +87,9 @@ def get_cloud_info_for_cluster(cls, kusto_uri: str, proxies: Optional[Dict[str,
             url = urljoin(kusto_uri, METADATA_ENDPOINT)
 
             # trace http get call for result
-            http_trace_attributes = KustoTracingAttributes.create_http_attributes(url=url, method="GET")
+            http_trace_attributes = KustoTracingAttributes.create_http_attributes(url=url, method="GET", max_redirects=0)

```

Comment:
We do not (currently) pass the client to cloudsettings. doing this would be a big change in architecture here. I think it's out of scope for this PR, unless you want to postpone it until possible